Dockerfile

```diff
@@ -22,36 +22,73 @@
 # from your docker directory.
 
 ARG BASE_IMAGE=ubuntu:22.04
-FROM $BASE_IMAGE AS builder
+
+
+######################################################################
+# A base image for building wheels
+FROM --platform=${TARGETPLATFORM:-linux/amd64} $BASE_IMAGE AS base-builder
 
 ENV DEBIAN_FRONTEND=noninteractive
 WORKDIR /src/jupyterhub
 
+# Ubuntu 22.04 comes with Nodejs 12 which is too old for building JupyterHub JS
+# It's fine at runtime though (used only by configurable-http-proxy)
 RUN apt update -q \
  && apt install -yq --no-install-recommends \
     build-essential \
     ca-certificates \
+    curl \
     locales \
     python3-dev \
     python3-pip \
     python3-pycurl \
     python3-venv \
+ && curl -fsSL https://deb.nodesource.com/setup_18.x | bash - \
+ && apt-get install -yq --no-install-recommends \
     nodejs \
-    npm \
  && apt clean \
  && rm -rf /var/lib/apt/lists/* \
  && python3 -m pip install --no-cache-dir --upgrade setuptools pip build wheel \
  && npm install --global yarn
+
 # copy everything except whats in .dockerignore, its a
 # compromise between needing to rebuild and maintaining
 # what needs to be part of the build
 COPY . .
+
+
+######################################################################
+# The JupyterHub wheel is pure Python so can be built once on any architecture
+# TODO: re-enable BUILDPLATFORM instead of forcing to linux/amd64
+#FROM --platform=${BUILDPLATFORM:-linux/amd64} base-builder AS jupyterhub-builder
+FROM --platform=linux/amd64 base-builder AS jupyterhub-builder
+
+# Need ARG to be able to view values
+# https://docs.docker.com/engine/reference/builder/#automatic-platform-args-in-the-global-scope
+ARG BUILDPLATFORM
+ARG TARGETPLATFORM
+RUN echo "BUILDPLATFORM=${BUILDPLATFORM} TARGETPLATFORM=${TARGETPLATFORM}"
```

Comment:
on linux/arm64:
```
2023-05-06T00:00:12.2369673Z #24 [linux/arm64 jupyterhub-builder 1/2] RUN echo "BUILDPLATFORM=linux/amd64 TARGETPLATFORM=linux/arm64"
2023-05-06T00:00:12.2370126Z #0 0.088 BUILDPLATFORM=linux/amd64 TARGETPLATFORM=linux/arm64
2023-05-06T00:00:12.3870583Z #24 DONE 0.1s
```
as expected..... so the JupyterHub build should be fast, but it's as if it's still being done on emulated arm64 :disappointed: